src/Messaging/src/RabbitMQ/Extensions/RabbitServicesExtensions.cs

```diff
@@ -244,17 +246,42 @@ public static IServiceCollection ConfigureRabbitOptions(this IServiceCollection
                 .Bind(config.GetSection(RabbitOptions.PREFIX))
                 .Configure<IServiceProvider>((options, provider) =>
                 {
-                    var connectionFactory = provider.GetService<RC.IConnectionFactory>() as RC.ConnectionFactory;
-
-                    if (connectionFactory is not null)
+                    using (var scope = provider.CreateScope())
                     {
-                        options.Addresses = $"{connectionFactory.HostName}:{connectionFactory.Port}";
+                        var connectionFactory = scope.ServiceProvider.GetService<RC.IConnectionFactory>() as RC.ConnectionFactory;
+
+                        if (connectionFactory is not null)
+                        {
+                            options.Addresses = $"{connectionFactory.UserName}:{connectionFactory.Password}@{connectionFactory.HostName}:{connectionFactory.Port}";
+                            options.VirtualHost = connectionFactory.VirtualHost;
+                            options.Host = connectionFactory.HostName;
+                            options.Username = connectionFactory.UserName;
+                            options.Password = connectionFactory.Password;
+                        }
                     }
                 });
 
             return services;
         }
 
+        public static IServiceCollection ConfigureRabbitOptionsForCloudFoundry(this IServiceCollection services, IConfiguration config)
```

Comment:
Is this an alternative to `ConfigureRabbitOptions` or might there be some case where they are used together?

I'm not sure that `ForCloudFoundry` is ideal here as settings in `rabbitmq:client` would also be picked up... What about `ConfigureRabbitOptionsFromConnectors`?

Adding xml comments to all public methods in this class would be a really good idea and help clarify questions like the above.

Reply:
This needs to be removed. Thanks!